### ISSUE TITLE
Angular (1.5) HTML attributes stop working when using FA5 svg-with-js

### ISSUE THREAD

**Issue by InFerYes** [FortAwesome/Font-Awesome#12055]
When I use a click event on an icon, the event will never fire because the original code is commented and replaced with SVG, for example:

`<span class="fas fa-cog" ng-click="controller.doSomething('withSomething')"></span>`

We have a very large application and the implications of this issue means that we not only need to replace/update any icons, but we also have to wrap them and move these events outside of the span, which drastically increases the time to update from 4 to 5. Using the webfonts version gives us no problems, but we miss out on all the svg-with-js features.

Another example that works with FA4 and FA5 webfonts is, but not svg-with-js:

`<i class="fas" ng-class="{'fa-chevron-up': status.opena, 'fa-chevron-down': !status.opena}"></i>`

Here the correct icon is shown based on the status of an accordion (open or closed), but the code never fires, because the conditional is commented.

I'm sure others have run into this issue, and I'm looking for advice to make the change as painless as possible. I have found [this](https://github.com/FortAwesome/angular-fontawesome/) but it's made for the newer angular version and not ready for production.

FontAwesome Pro 5.0.2
Angular 1.5.9

**Comment by yoat** [comment]
I am using AngularJS 1.6 and what you describe isn't an issue.
Yes, the <i> gets commented out, but the ng-class property does get copied over to the SVG:

`
<svg class="svg-inline--fa fa-file-pdf fa-w-16 ng-animate" ng-class="file.Text | fileExtension | fileIconFa" data-ng-animate="2" aria-hidden="true" data-fa-processed="" data-prefix="fas" data-icon="file-pdf" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
...
</svg>

`

Are you sure the ng-class isn't being duplicated into the SVG? I am having different issues with how that ng-class is getting processed, but it is executing my code at least.

**Comment by yoat** [comment]
I noticed something about my previous comment: the ng-class got rendered in the `<i>` before it got commented out, and then both the ng-class and the class were getting copied over to the `<svg>`.

My problem (not previously mentioned) was the (something) was occasionally appending "-add" to the "fa-file-pdf" class and the data-icon value, which showed up as an unknown icon.  

After realizing that the class was being rendered by AngularJS before FA was replacing the I with an SVG gave me an idea, so I changed the template from
`<i class="fas" ng-class="file.Text | fileExtension | fileIconFa"></i>`
to
`<i class="fas {{file.Text | fileExtension | fileIconFa}}"></i>`

And the SVG tag went from
`<svg class="svg-inline--fa fa-file-pdf fa-w-16 ng-animate" ng-class="file.Text | fileExtension | fileIconFa" data-ng-animate="2" aria-hidden="true" data-fa-processed="" data-prefix="fas" data-icon="file-pdf"...`
to
`<svg class="svg-inline--fa fa-file-pdf fa-w-12" aria-hidden="true" data-fa-processed="" data-prefix="fas" data-icon="file-pdf" ...`

In short, switching from using ng-class to just rendering the class name inline with {{}} resolved my issue and may help with yous.

**Comment by InFerYes** [comment]
Yes, this seems to be a solution. The ng-class conditional that is copied over are designed to work with `i` or `span`, but not `svg` tags, that's probably why they fail. This will be quite the effort, but at least we can make it work.

**Comment by InFerYes** [comment]
Here is another example of code that stops working. The attributes are copied over, but doesn't function anymore.

```HTML
<svg xmlns="http://www.w3.org/2000/svg" class="svg-inline--fa fa-shield-check fa-w-16 handCursor" role="img" aria-hidden="true" viewBox="0 0 512 512" popover-placement="top" uib-popover-template="'tplRollen.html'" data-icon="shield-check" data-prefix="fas" data-fa-processed=""><path fill="currentColor" d="M 496 128 c 0 221.282 -135.934 344.645 -221.539 380.308 a 48 48 0 0 1 -36.923 0 C 130.495 463.713 16 326.487 16 128 a 48 48 0 0 1 29.539 -44.308 l 192 -80 a 48 48 0 0 1 36.923 0 l 192 80 A 48 48 0 0 1 496 128 Z M 235.313 381.941 l 184 -184 c 6.248 -6.248 6.248 -16.379 0 -22.627 l -22.627 -22.627 c -6.248 -6.248 -16.379 -6.249 -22.628 0 L 224 302.745 l -70.059 -70.059 c -6.248 -6.248 -16.379 -6.248 -22.628 0 l -22.627 22.627 c -6.248 6.248 -6.248 16.379 0 22.627 l 104 104 c 6.248 6.25 16.379 6.25 22.627 0.001 Z" /></svg>
<!-- <span class="fas fa-shield-check handCursor" popover-placement="top" uib-popover-template="'tplRollen.html'"></span> --> 
```

**Comment by ghs** [comment]
Yes, same here. Just switched to try v.5 and the first thing that happened was that angular directives containing logic are just copied and making them unavailable.

**Comment by sensibleworld** [comment]
Doing a little management in Github and (batch) closing out potentially stale issues. If the above still needs attention, please let us know and we’ll re-open!
